src/simple_graph.py

```diff
@@ -0,0 +1,94 @@
+"""An unweighted, directed graph."""
+import random
+import string
+
+
+class SimpleGraph(object):
+    """Simple Graph Object."""
+
+    def __init__(self):
+        """Initiate an empty graph."""
+        self.graph = {}
+
+    def add_node(self, val):
+        """Add a node to the graph."""
+        self.graph.setdefault(val, [])
+
+    def add_edge(self, val, edge):
+        """Add value and it's edge."""
+        if edge not in self.graph:
+            self.add_node(edge)
+
+        if val in self.graph:
+            if edge not in self.graph[val]:
+                self.graph[val].append(edge)
+        else:
+            self.graph[val] = [edge]
+
+    def nodes(self):
+        """Return all nodes in the graph."""
+        return list(self.graph.keys())
+
+    def edges(self):
+        """Return all edges in the graph."""
+        all_edges = []
+        for node in self.graph:
+            for edge in self.graph[node]:
+                all_edges.append((node, edge))
+
+        return all_edges
+
+    def del_node(self, val):
+        """If node in graph, delete it."""
+        try:
+            val in self.graph
+            for loop_val in self.graph:
+                if val in self.graph[loop_val]:
+                    self.graph[loop_val].remove(val)
+            del self.graph[val]
+        except KeyError:
+            raise ValueError("Node doesn't exist")
+
+    def del_edge(self, val, edge):
+        """Remove connection between node and edge if edge exists."""
+        try:
+            val in self.graph
+            try:
+                edge in self.graph[val]
+                self.graph[val].remove(edge)
+            except ValueError:
+                raise ValueError("Edge doesn't exist")
+        except KeyError:
+            raise ValueError("Node doesn't exist")
+
+    def has_node(self, val):
+        """If node exists return True."""
+        return val in self.graph
+
+    def neighbors(self, val):
+        """Return all edges of a node."""
+        if val not in self.graph:
+            raise ValueError("Node doesn't exist.")
+
+        return self.graph[val]
+
+    def adjacent(self, val1, val2):
+        """Return if one node has edge with the other."""
+        if val1 not in self.graph or val2 not in self.graph:
+            raise ValueError("Node doesn't exist.")
+
+        return val1 in self.graph and val2 in self.graph[val1]
+
+    def random_ascii_char(self):  # pragma: no cover
+        """Return random upper case ascii char between 'A' and  'J'."""
+        return random.choice(string.ascii_uppercase[:10])
+
+    def create_random_graph(self, nodes=20):  # pragma no cover
```

Comment:
If you want to build a constructor for testing, then build a function that creates a random graph for you. This shouldn't be a method of a graph object. Or make it a `@classmethod`